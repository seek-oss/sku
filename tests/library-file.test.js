const path = require('node:path');
const {
  dirContentsToObject,
  runSkuScriptInDir,
  waitForUrls,
  getAppSnapshot,
} = require('@sku-private/test-utils');

const appDir = path.dirname(
  require.resolve('@sku-fixtures/library-file/sku.config.js'),
);
const distDir = path.resolve(appDir, 'dist');

describe('library-file', () => {
  describe('build', () => {
    beforeAll(async () => {
      await runSkuScriptInDir('build', appDir);
    });

    it('should generate the expected files', async () => {
      const files = await dirContentsToObject(distDir);
      expect(files).toMatchSnapshot();
    });
  });

  describe('start', () => {
    const devServerUrl = `http://localhost:8086`;
    let server;

    beforeAll(async () => {
      server = await runSkuScriptInDir('start', appDir);
      await waitForUrls(devServerUrl);
    });

    afterAll(async () => {
      await server.kill();
    });

    it('should start a development server', async () => {
      const snapshot = await getAppSnapshot(devServerUrl);
      expect(snapshot).toMatchSnapshot();
    });
  });
});
