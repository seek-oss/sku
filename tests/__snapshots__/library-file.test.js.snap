// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`library-file build should generate the expected files 1`] = `
{
  "161-c28a27d2bffe342554e3.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "663-908832b5cabbdf610215.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "my-library.css": .LfCbsbg {
  color: red;
}
.MyLibrary_llPTb {
  background-color: #20b2aa;
  color: #fff;
  font-size: 50px;
  padding: 20px;
}
,
  "my-library.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
}
`;

exports[`library-file start should start a development server 1`] = `
SCRIPTS: [
  "/my-library.js",
]
CSS: [
  "/my-library.css",
]
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      My Awesome Project
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <link
      rel="stylesheet"
      type="text/css"
      href="styles[0]"
    >
  </head>
  <body>
    <script src="scripts[0]">
    </script>
    <script>
      window.MyLibrary();
    </script>
  </body>
</html>
POST HYDRATE DIFFS: 
     <script>
       window.MyLibrary();
     </script>
+    <div class="styles__root___LfCbsbg MyLibrary_BRAID__styles-customBox_llPTb">
+      This element was generated by the library
+    </div>
   </body>
 </html>
\\ No newline at end of file

`;
